underfs/s3/src/main/java/alluxio/underfs/s3/S3UnderFileSystemFactory.java

```diff
@@ -39,20 +39,23 @@
   public UnderFileSystem create(String path, Configuration configuration, Object unusedConf) {
     Preconditions.checkNotNull(path);
     Preconditions.checkNotNull(configuration);
+    AWSCredentials creds;
 
     if (addAndCheckAWSCredentials(configuration)) {
-      AlluxioURI uri = new AlluxioURI(path);
-      try {
-        return new S3UnderFileSystem(uri.getHost(), configuration);
-      } catch (ServiceException e) {
-        LOG.error("Failed to create S3UnderFileSystem.", e);
-        throw Throwables.propagate(e);
-      }
+      creds = new AWSCredentials(configuration.get(Constants.S3_ACCESS_KEY),
+              configuration.get(Constants.S3_SECRET_KEY));
+    } else {
+      creds = AWSEC2IAMSessionCredentials.loadFromEC2InstanceData(true);
```

Comment:
How should we handle this case ? Do you think if it makes sense to introduce a new config/flag which is something like `fs.s3n.useInstanceRole=true` ? If this is set to false(which is by default) and instance is not running on AWS or running without a role, we can throw an exception saying `neither creds nor role provided`. Does this approach makes sense ?